pkg/common/cns-lib/volume/manager.go

```diff
@@ -110,7 +116,7 @@ func ClearTaskInfoObjects() {
 }
 
 // CreateVolume creates a new volume given its spec.
-func (m *volumeManager) CreateVolume(spec *cnstypes.CnsVolumeCreateSpec) (*cnstypes.CnsVolumeId, error) {
+func (m *volumeManager) CreateVolume(spec *cnstypes.CnsVolumeCreateSpec, storeCreateVolumeTaskInCache bool) (*cnstypes.CnsVolumeId, error) {
```

Comment:
Can you keep the existing signature and decide about caching or look up tasks based on the `CnsVolumeCreateSpec`. 

If BakingDiskID is present in the createspec, means volume is requested statically, so we do not need to look into the cache and we can also skip storing task for such volume creation in the cache, because static volume creation is never going to create any orphan volume due to timed out long-running tasks.

Please handle this for both file volume and block volume.

Reply:
Done.

We do not have file volume support in 1.0 and `CnsFileBackingDetail` in `cnstypes` as well. So we are good there